packages/starlight/style/props.css

```diff
@@ -79,7 +79,7 @@
 	--sl-text-h4: var(--sl-text-xl);
 	--sl-text-h5: var(--sl-text-lg);
 
-	--sl-line-height: 1.8;
+	--sl-line-height: 1.75;
```

Comment:
This change is on purpose to avoid sub-pixel scroll position changes when switching between tabs with different content height.

In some browsers (Chrome (Blink) and Firefox), a line height with a fractional part (which is the case of Starlight before this change: `16 * 1.8 = 28.8`) seems to cause scripts accessing dimensions involving the line height to get an inconsistent rounded value.
In the specific case of tabs, when restoring the scroll position after switching between tabs and especially when the content height is different between tabs, this could cause a sub-pixel scroll position change.

https://github.com/withastro/starlight/assets/494699/26218f0c-79c8-416b-96b8-4783d2190755

This change brings the base line height to a whole number (`16 * 1.75 = 28`) which avoids the issue in all browsers I tested and all device pixel densities I could test.

https://github.com/withastro/starlight/assets/494699/c053b6e6-9511-499b-9c10-b7baf172e757

I also added a test which ensure the bounding box of a set of tabs is identical before and after switching tabs in a case where a tiny scroll position would have been applied before.

---

Additional question: I guess this should have its own changeset, right, as subtle as it is?

Reply:
Yes, I agree it might be a good idea to add a separate changeset documenting the line-height change, why it was made, and how to revert it if someone really wants to.